packages/flutter_tools/templates/template_manifest.json

```diff
@@ -349,6 +349,51 @@
         "templates/plugin/windows.tmpl/pluginClassSnakeCase.cpp.tmpl",
         "templates/plugin/lib/projectName_web.dart.tmpl",
 
+        "templates/ffi_plugin/.gitignore.tmpl",
+        "templates/ffi_plugin/.idea/libraries/Dart_SDK.xml.tmpl",
+        "templates/ffi_plugin/.idea/modules.xml.tmpl",
+        "templates/ffi_plugin/.idea/runConfigurations/example_lib_main_dart.xml.tmpl",
+        "templates/ffi_plugin/.idea/workspace.xml.tmpl",
+        "templates/ffi_plugin/.metadata.tmpl",
+        "templates/ffi_plugin/analysis_options.yaml.tmpl",
+        "templates/ffi_plugin/android.tmpl/.gitignore",
+        "templates/ffi_plugin/android.tmpl/build.gradle.tmpl",
+        "templates/ffi_plugin/android.tmpl/gradle/wrapper/gradle-wrapper.properties",
+        "templates/ffi_plugin/android.tmpl/gradle.properties.tmpl",
+        "templates/ffi_plugin/android.tmpl/settings.gradle.tmpl",
+        "templates/ffi_plugin/android.tmpl/projectName_android.iml.tmpl",
+        "templates/ffi_plugin/android.tmpl/src/main/AndroidManifest.xml.tmpl",
+        "templates/ffi_plugin/CHANGELOG.md.tmpl",
+        "templates/ffi_plugin/ios.tmpl/.gitignore",
+        "templates/ffi_plugin/ios.tmpl/Assets/.gitkeep",
+        "templates/ffi_plugin/ios.tmpl/Classes/dart_api_dl.c",
+        "templates/ffi_plugin/ios.tmpl/Classes/projectName.c.tmpl",
+        "templates/ffi_plugin/ios.tmpl/projectName.podspec.tmpl",
+        "templates/ffi_plugin/lib/projectName.dart.tmpl",
+        "templates/ffi_plugin/LICENSE.tmpl",
+        "templates/ffi_plugin/linux.tmpl/CMakeLists.txt.tmpl",
+        "templates/ffi_plugin/linux.tmpl/include/projectName.tmpl/ffi_pluginClassSnakeCase.h.tmpl",
+        "templates/ffi_plugin/macos.tmpl/Classes/dart_api_dl.c",
+        "templates/ffi_plugin/macos.tmpl/Classes/projectName.c.tmpl",
+        "templates/ffi_plugin/macos.tmpl/projectName.podspec.tmpl",
+        "templates/ffi_plugin/projectName.iml.tmpl",
+        "templates/ffi_plugin/pubspec.yaml.tmpl",
+        "templates/ffi_plugin/README.md.tmpl",
+        "templates/ffi_plugin/src.tmpl/dart-sdk/include/internal/dart_api_dl_impl.h",
+        "templates/ffi_plugin/src.tmpl/dart-sdk/include/dart_api_dl.c",
+        "templates/ffi_plugin/src.tmpl/dart-sdk/include/dart_api_dl.h",
+        "templates/ffi_plugin/src.tmpl/dart-sdk/include/dart_api.h",
+        "templates/ffi_plugin/src.tmpl/dart-sdk/include/dart_native_api.h",
+        "templates/ffi_plugin/src.tmpl/dart-sdk/include/dart_tools_api.h",
+        "templates/ffi_plugin/src.tmpl/dart-sdk/include/dart_version.h",
```

Comment:
is there any way we can import these from the Dart SDK rather than copying them? It seems really sketchy to copy these files verbatim into plugins...

Reply:
I'll remove `dart_api_dl.h` from this PR and address it in a follow up PR after we address https://github.com/dart-lang/sdk/issues/47649 and the other issues raised in the API review.